### PR TITLE
DIT-2083 convert JSON upload to stream JSON rather than read the whole file

### PR DESCRIPTION
**Pull request by DavidGregory084**

### CODE REVIEW

File: app/uk/gov/hmrc/bindingtariffadminfrontend/controllers/CaseMigrationUploadController.scala [hunk 1]
```diff
@@ -59,45 +73,51 @@ class CaseMigrationUploadController @Inject()(authenticatedAction: Authenticated
...
-          case JsError(errs) => successful(Ok(views.html.case_migration_file_error(errs)))
-          case JsSuccess(migrations, _) =>
-            service.prepareMigration(migrations, priority).map(_ => Redirect(routes.DataMigrationStateController.get()))
-        }
+        service.prepareMigration(toJsonSource(part.ref.file, part.contentType), priority)
+          .map(_ => Redirect(routes.DataMigrationStateController.get()))
+          .recoverWith {
+            // Happens when Akka doesn't know how to split the file into JSON chunks
+            case framing: FramingException =>
+              val indexZero = JsPath.apply(0)
```
Comment (DavidGregory084) [inline app/uk/gov/hmrc/bindingtariffadminfrontend/controllers/CaseMigrationUploadController.scala:81]: Note to self - change this to remove `.apply`

Comment (colesadam) [reply app/uk/gov/hmrc/bindingtariffadminfrontend/controllers/CaseMigrationUploadController.scala:81]: You still doing this?

Comment (colesadam) [reply app/uk/gov/hmrc/bindingtariffadminfrontend/controllers/CaseMigrationUploadController.scala:81]: I'm merging it :)

File: test/unit/uk/gov/hmrc/bindingtariffadminfrontend/controllers/CaseMigrationUploadControllerSpec.scala [hunk 2]
```diff
@@ -78,52 +156,38 @@ class CaseMigrationUploadControllerControllerSpec extends WordSpec with Matchers
...
-      given(migrationService.prepareMigration(any[Seq[MigratableCase]], refEq(true))(any[HeaderCarrier])) willReturn Future.successful(true)
+    // TODO: Fix these tests! When we changed this suite we found that they do not test anything;
+    // they mocked the main method that this controller calls in DataMigrationService so all of
+    // the different inputs provided are totally irrelevant as none of the associated logic is
+    // ever called.
+    //
+    // When we changed the test suite so that we no longer mock the DataMigrationService we discovered
+    // that to make these tests work we would need to mock responses from a bunch of other services,
+    // such as the FileStoreConnector, CaseConnector and RulingConnector.
+    //
```
Comment (DavidGregory084) [inline test/unit/uk/gov/hmrc/bindingtariffadminfrontend/controllers/CaseMigrationUploadControllerSpec.scala:186]: I have raised DIT-2131 to capture this

**Pull request merged**